File: Ex2_calc.swift
// Calculator class representing a basic calculator with add, subtract, multiply, and divide operations
class Calculator {

    // Function to add two numbers
    func add(_ a: Double, _ b: Double) -> Double {
        return a + b
    }

    // Function to subtract two numbers
    func subtract(_ a: Double, _ b: Double) -> Double {
        return a - b
    }

    // Function to multiply two numbers
    func multiply(_ a: Double, _ b: Double) -> Double {
        return a * b
    }

    // Function to divide two numbers
    func divide(_ a: Double, _ b: Double) -> Double {
        guard b != 0 else {
            fatalError("Cannot divide by zero")
        }
        return a / b
    }
}

// Instantiate the Calculator class
let calculator = Calculator()

// Perform some calculations
let resultAdd = calculator.add(10.0, 5.0)
print("10 + 5 = \(resultAdd)")

let resultSubtract = calculator.subtract(10.0, 5.0)
print("10 - 5 = \(resultSubtract)")

let resultMultiply = calculator.multiply(10.0, 5.0)
print("10 * 5 = \(resultMultiply)")

let resultDivide = calculator.divide(10.0, 5.0)
print("10 / 5 = \(resultDivide)")

// Trying to divide by zero will cause a fatal error
// Uncomment the following line to see the error
// let resultDivideByZero = calculator.divide(10.0, 0.0)

File: Ex2_calc.kt
// Calculator class representing a basic calculator with add, subtract, multiply, and divide operations
class Calculator {

    // Function to add two numbers
    fun add(a: Double, b: Double): Double {
        return a + b
    }

    // Function to subtract two numbers
    fun subtract(a: Double, b: Double): Double {
        return a - b
    }

    // Function to multiply two numbers
    fun multiply(a: Double, b: Double): Double {
        return a * b
    }

    // Function to divide two numbers
    fun divide(a: Double, b: Double): Double {
        if (b == 0.0) {
            throw IllegalArgumentException("Cannot divide by zero")
        }
        return a / b
    }
}

fun main() {
    val calculator = Calculator()

    // Perform some calculations
    val resultAdd = calculator.add(10.0, 5.0)
    println("10 + 5 = $resultAdd")

    val resultSubtract = calculator.subtract(10.0, 5.0)
    println("10 - 5 = $resultSubtract")

    val resultMultiply = calculator.multiply(10.0, 5.0)
    println("10 * 5 = $resultMultiply")

    val resultDivide = calculator.divide(10.0, 5.0)
    println("10 / 5 = $resultDivide")

    // Trying to divide by zero will throw an exception
    try {
        val resultDivideByZero = calculator.divide(10.0, 0.0)
        println("10 / 0 = $resultDivideByZero")
    } catch (e: IllegalArgumentException) {
        println("Error: ${e.message}")
    }
}
